### PR TITLE
add recovery_wait_secs option for MonitoredTrainingSession

### PR DESCRIPTION
**Pull request by horance-liu**
In order to reduce the sleep time by worker to wait for a model to be initialized or restored , 
，we add  recovery_wait_secs option for MonitoredTrainingSession，and we are able to start distributed training faster.

### CODE REVIEW

**Comment by frankchn** [comment]
Jenkins, test this please.

**Comment by jhseu** [comment]
Let's wait for @ispirmustafa to return to review, since he owns this class. Might take a few weeks.

**Comment by rmlarsen** [comment]
@ispirmustafa could you take a look, please?

File: tensorflow/python/training/monitored_session.py
Comment (ispirmustafa) [inline tensorflow/python/training/monitored_session.py]: here it needs better documentation. we should mention that this is used by worker jobs to check whether chief initialized/restored session or not.

**Review by ispirmustafa — CHANGES_REQUESTED** [review]
Thank you!
One small doc change, otherwise LGTM.

File: tensorflow/python/training/monitored_session.py
Comment (horance-liu) [inline tensorflow/python/training/monitored_session.py]: Thanks, I have appended more comment for this parameter.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Review by ispirmustafa — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@martinwicke looks like this is good to go, pending API review.

File: tensorflow/python/training/monitored_session.py
Comment (martinwicke) [inline tensorflow/python/training/monitored_session.py]: remove "this"?

**Review by martinwicke — DISMISSED** [review]
One little docs change, otherwise good.

**Comment by martinwicke** [comment]
LG for API once docstring is fixed.

File: tensorflow/python/training/monitored_session.py
Comment (horance-liu) [inline tensorflow/python/training/monitored_session.py]: I'm sorry, I have fixed it.

**Comment by horance-liu** [comment]
@rmlarsen I have fixed it for comment.

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by jhseu** [comment]
Mind updating the goldens? See the failing test. Thanks!

**Comment by horance-liu** [comment]
I run the test as follows to update test goldens, but nothing changed.

    $ bazel build tensorflow/tools/api/tests:api_compatibility_test
    $ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True

image

What should I do next?  Thanks

**Comment by martinwicke** [comment]
@gunan Can you see what went wrong here?

**Comment by gunan** [comment]
You need to run the api compatibility test with python 2. Are you using python3, by any chance?

**Comment by martinwicke** [comment]
We should probably point that out explicitly in the failure message if it
doesn't already.

**Comment by drpngx** [comment]
FTR, the error is:
```
ERROR:tensorflow:2 differences found between API and golden.
ERROR:tensorflow:Issue 1	: Change detected in python object: tensorflow.train.WorkerSessionCreator.
ERROR:tensorflow:Issue 2	: Change detected in python object: tensorflow.train.
F.
```

@horance-liu did you run the golden generator? Did you run it in python2 or python3?

**Comment by sb2nov** [comment]
@horance-liu ping

**Comment by horance-liu** [comment]
I have run the golden generator in python2, but nothing changed. See [](https://user-images.githubusercontent.com/13819195/29695085-27a0b376-8973-11e7-94e3-3c3266487e09.png)

**Comment by horance-liu** [comment]
I have run the golden generator in python2, but nothing changed. See https://user-images.githubusercontent.com/13819195/29695085-27a0b376-8973-11e7-94e3-3c3266487e09.png

**Comment by gunan** [comment]
Sorry for incomplete instructions. You need to be on Linux, and python2 to regenerate goldens. I will look into what we need to also include windows and macos in the OS side.
If you are working on a macos system, you may use docker to do the golden update.

**Comment by martinwicke** [comment]
@asimshankar @josh11b This is the same change we just reviewed... we said back then it was ok. Didn't make the connection until now. Same question applies -- why is there a timeout at all?

**Comment by martinwicke** [comment]
I think this has since been independently added (albeit with a radically different default). I'll close this PR.

**Pull request closed without merging**
